Compile bug: Can no longer configure CMake on MSVC with CMP0194 set to NEW

### Git commit

4f02d4733934179386cbc15b3454be26237940bb

### Operating systems

Windows

### GGML backends

CUDA

### Problem description & steps to reproduce

Setting the CMake policy CMP0194 to NEW requires that an ASM compiler be present, and cmake configure now fails, since MSVC is no longer considered an ASM compiler by newer CMake verisons.
Removing the policy line makes it configure again, albeit with the warning the offending commit intended to remove.

I think the correct fix is to only enable ASM with [enable_language](https://cmake.org/cmake/help/latest/command/enable_language.html) when one of the backends that use it (per the commit message in the breaking change) are enabled

### First Bad Commit

acc37a42eaba02e57065fef3b415d9f1e64018fa

### Compile command

```shell
cmake -B build -DGGML_CUDA=ON
```

### Relevant log output

```shell
-- The ASM compiler identification is unknown
-- Didn't find assembler
CMake Error at ggml/CMakeLists.txt:9 (project):
  No CMAKE_ASM_COMPILER could be found.
```

--- a/ggml/CMakeLists.txt
+++ b/ggml/CMakeLists.txt
@@ -1,11 +1,5 @@
 cmake_minimum_required(VERSION 3.14...3.28) # for add_link_options and implicit target directories.
 
-# ref: https://cmake.org/cmake/help/latest/policy/CMP0194.html
-# MSVC is not a valid assembler for the ASM language.
-# Set to NEW to avoid a warning on CMake 4.1+ with MSVC.
-if (POLICY CMP0194)
-    cmake_policy(SET CMP0194 NEW)
-endif()
 project("ggml" C CXX ASM)
 
 ### GGML Version

